src/sparse_merkle_tree/smt.rs

```diff
@@ -1538,7 +1538,7 @@ impl SparseMerkleTree {
 
         // Check if all the query keys have the same length.
         for query in &proof.queries {
-            if query.pair.key().len() != key_length.into() {
+            if query.key().len() != key_length.into() || query.key().len() < query.bitmap.len() {
```

Comment:
i think here you need to compare with `binary_bitmap`. At this point `query.bitmap` is still in bytes format

Reply:
Done.